src/main/java/org/gridsuite/timeseries/server/TimeseriesDataRepository.java

```diff
@@ -0,0 +1,347 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.timeseries.server;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ForkJoinTask;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Repository;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.powsybl.timeseries.DoubleDataChunk;
+import com.powsybl.timeseries.DoubleTimeSeries;
+import com.powsybl.timeseries.StoredDoubleTimeSeries;
+import com.powsybl.timeseries.StringDataChunk;
+import com.powsybl.timeseries.StringTimeSeries;
+import com.powsybl.timeseries.TimeSeries;
+import com.powsybl.timeseries.TimeSeriesIndex;
+import com.powsybl.timeseries.UncompressedDoubleDataChunk;
+import com.powsybl.timeseries.UncompressedStringDataChunk;
+import com.zaxxer.hikari.HikariDataSource;
+
+/**
+ * @author Jon Schuhmacher <jon.harper at rte-france.com>
+ */
+@Repository
+public class TimeseriesDataRepository {
```

Comment:
In the future, in my opinion, we can extract all technical codes which uses multithreads into an under layer, e.g. a DAO class which processes in batch mode for CRUD operations.